packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -517,17 +543,26 @@ export class TeamsBotImpl {
   public async deploy(context: PluginContext): Promise<FxResult> {
     this.ctx = context;
     await this.config.restoreConfigFromContext(context);
+
+    this.config.provision.subscriptionId = getSubscriptionIdFromResourceId(
+      this.config.provision.botWebAppResourceId!
+    );
+    this.config.provision.resourceGroup = getResourceGroupNameFromResourceId(
+      this.config.provision.botWebAppResourceId!
+    );
+
```

Comment:
we also need to extract the web app name from resource id?

Reply:
Yes. Fixed.